swap/simulations_test.go

```diff
@@ -0,0 +1,779 @@
+			if len(debitorSvc.swap.peers) == 1 && len(creditorSvc.swap.peers) == 1 {
+				break
+			}
+			// don't overheat the CPU...
+			time.Sleep(5 * time.Millisecond)
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		// we will send just maxCheques number of cheques
+		maxCheques := 10
+
+		// the peer object used for sending
+		creditorPeer := debitorSvc.peers[creditor]
+
+		// send maxCheques number of cheques
+		for i := 0; i < maxCheques; i++ {
+			// use a price which will trigger a cheque each time
+			creditorPeer.Send(ctx, &testMsgBigPrice{})
+			// we need to wait a bit in order to give time for the cheque to be processed
+			err = waitForChequeProcessed(creditorSvc)
+			if err != nil {
+				return err
+			}
+		}
+
+		// check balances:
+		b1, err := debitorSvc.swap.loadBalance(creditor)
+		if err != nil {
+			return err
+		}
+		b2, err := creditorSvc.swap.loadBalance(debitor)
+		if err != nil {
+			return err
+		}
+
+		if b1 != -b2 {
+			return fmt.Errorf("Expected symmetric balances, but they are not: %d vs %d", b1, b2)
+		}
+		// check cheques
+		var cheque1, cheque2 *Cheque
+		if cheque1, err = debitorSvc.swap.loadLastSentCheque(creditor); err != nil {
+			return errors.New("expected cheques with creditor, but none found")
+		}
+		if cheque2, err = creditorSvc.swap.loadLastReceivedCheque(debitor); err != nil {
+			return errors.New("expected cheques with debitor, but none found")
+		}
+
+		// both cheques (at issuer and beneficiary) should have same cumulative value
+		if cheque1.CumulativePayout != cheque2.CumulativePayout {
+			return fmt.Errorf("Expected symmetric cheques payout, but they are not: %d vs %d", cheque1.CumulativePayout, cheque2.CumulativePayout)
+		}
+
+		// check also the actual expected amount
+		expectedPayout := uint64(maxCheques * (DefaultPaymentThreshold + 1))
+
+		if cheque2.CumulativePayout != expectedPayout {
+			return fmt.Errorf("Expected %d in cumulative payout, got %d", expectedPayout, cheque1.CumulativePayout)
+		}
+
+		return nil
+
+	})
+
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestSimpleSimulation starts 16 nodes, then in a simple round robin fashion sends messages to each other.
+// Then checks that accounting is ok. It checks the actual amount of balances without any cheques sent,
+// in order to verify that the most basic accounting works.
+func TestSimpleSimulation(t *testing.T) {
+	nodeCount := 16
+	// create the shared backend and params
+	params, err := newSharedBackendSwaps(nodeCount)
+	if err != nil {
+		t.Fatal(err)
+	}
+	// cleanup backend
+	defer params.backend.Close()
+
+	// initialize the simulation
+	sim := simulation.NewInProc(newSimServiceMap(params))
+	defer sim.Close()
+
+	log.Info("Initializing")
+
+	ctx, cancelSimRun := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancelSimRun()
+
+	_, err = sim.AddNodesAndConnectFull(nodeCount)
```

Comment:
i think you should use snapshot which is the natural connectivity and easier to manage (you dont need the for loop on line 535, etc

Reply:
We talked about this in the last sync call. We already store two snapshot files for each service area they are needed for: syncing and pss. 

As the snapshot contains the name of the service, we would have to produce another set (at least one file with 16 nodes for this test) just for the swap simulation. 

Also, for this test the actual network topology and configuration (if kademlia or not) is irrelevant, it just tests that accounting works right.

But if you insist in having a snapshot here, I can produce one and replace it.